qiskit/transpiler/passes/optimization/unroll_forloops.py

```diff
@@ -0,0 +1,78 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" UnrollForLoops transpilation pass """
+
+from qiskit.circuit import ForLoopOp, ContinueLoopOp, BreakLoopOp, IfElseOp
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.utils import control_flow
+from qiskit.converters import circuit_to_dag
+
+
+class UnrollForLoops(TransformationPass):
+    """UnrollForLoops transpilation pass unrolls for-loops when possible. Things like
+    `for x in {0, 3, 4} {rx(x) qr[1];}` will turn into `rx(0) qr[1]; rx(3) qr[1]; rx(4) qr[1];`.
+    """
+
+    def __init__(self, max_target_depth=-1):
+        """UnrollForLoops transpilation pass unrolls for-loops when possible. Things like
+        `for x in {0, 3, 4} {rx(x) qr[1];}` will turn into `rx(0) qr[1]; rx(3) qr[1]; rx(4) qr[1];`.
+
+        Args:
+            max_target_depth (int): Optional. Checks if the unrolled block is over a particular depth.
+                To disable the check, use ``-1`` (Default)
+        """
+        super().__init__()
+        self.max_target_depth = max_target_depth
+
+    @control_flow.trivial_recurse
+    def run(self, dag):
+        """Run the UnrollForLoops pass on `dag`.
+
+        Args:
+            dag (DAGCircuit): the directed acyclic graph to run on.
+
+        Returns:
+            DAGCircuit: Transformed DAG.
+        """
+        for forloop_op in dag.op_nodes(ForLoopOp):
+            (indexset, loop_parameter, body) = forloop_op.op.params
+
+            # skip unrolling if it results in bigger than max_target_depth
+            if self.max_target_depth > 0 and len(indexset) * body.depth() > self.max_target_depth:
```

Comment:
if you like, you can spell this as
```suggestion
            if 0 < self.max_target_depth < len(indexset) * body.depth():
```
it has the same short-circuit behaviour as the `and` form, and means (very nearly) the same thing. _Technically_ it also guarantees that `self.max_target_depth` is only evaluated once rather than twice, but that's irrelevant in this situation.

Reply:
Much nicer! Thanks!